File: python/ABSTRACT_FACTORY_DECORATOR/ABSTRACT_FACTORY_DECORATOR/decorator/ConcreteDecoratorReservacionMultiple.py
from ABSTRACT_FACTORY_DECORATOR.reservacion.ReservacionMultiple import ReservacionMultiple
from .ReservacionDecoratorFactory import ReservacionDecoratorFactory

class ConcreteDecoratorReservacionMultiple(ReservacionDecoratorFactory):
    
    reservacionExtendida : ReservacionMultiple
    
    def __init__(self, reservacionSencilla) -> None:
        super().__init__(reservacionSencilla)
        
    def aumentarReservacion(self):
        self.volverReservacionMultiple()
    
    def volverReservacionMultiple(self):
        self.reservacionExtendida = ReservacionMultiple()
        self.reservacionExtendida.agregarReservacion(self.reservacionSencilla)

File: php/ABSTRACT_FACTORY_DECORATOR/decorator/ConcreteDecoratorReservacionMultiple.php
<?php

require_once 'ReservacionDecoratorFactory.php';
require_once './reservacion/ReservacionMultiple.php';

class ConcreteDecoratorReservacionMultiple extends ReservacionDecoratorFactory{
    public $reservacionExtendida;
    
    public function __construct($reservacionSencilla) {
        parent::__construct($reservacionSencilla);
    }
    
    public function aumentarReservacion($datos){
        $this->volverReservacionMultiple();
    }
    
    public function volverReservacionMultiple(){
        $this->reservacionExtendida = new ReservacionMultiple();
        $this->reservacionExtendida->agregarReservacion($this->reservacionSencilla);
    }
}
